test/e2e/all_in_one_test.go

```diff
@@ -475,3 +571,81 @@ func exposeAdminAPI(ctx context.Context, t *testing.T, env environments.Environm
 
 	return service
 }
+
+// getTestManifestPath checks if a controller image override is set. If not, it returns the original provided path.
+// If an override is set, it runs a kustomize patch that replaces the controller image with the override image and
+// returns the modified manifest path. If there is any issue patching the manifest, it will log the issue and return
+// the original provided path
+func getTestManifestPath(t *testing.T, baseManifestPath string) string {
```

Comment:
would it be possible to make this patching work without saving files on the filesystem (which are then either leaked or need to be cleaned up specififcally?

Reply:
bb47ebb0, though it'll get annoying to debug if we start doing more complex operations.